river/Cursor.zig

```diff
@@ -185,15 +186,43 @@ const Mode = union(enum) {
 
     /// Pass an event on to the surface under the cursor, if any.
     fn passthrough(self: *Self, time: u32) void {
+        // Get the output where the cursor is
+        const root = &self.seat.input_manager.server.root;
+        const config = self.seat.input_manager.server.config;
+
+        var ox = self.wlr_cursor.x;
+        var oy = self.wlr_cursor.y;
+        // TODO should we rellly return here? Maybe catch unreachable
+        const output = Output.outputAt(root, &ox, &oy) orelse return;
+
         var sx: f64 = undefined;
         var sy: f64 = undefined;
-        if (self.surfaceAt(self.wlr_cursor.x, self.wlr_cursor.y, &sx, &sy)) |wlr_surface| {
+        if (output.surfaceAt(ox, oy, &sx, &sy)) |wlr_surface| {
             // If input is allowed on the surface, send pointer enter and motion
             // events. Note that wlroots won't actually send an enter event if
             // the surface has already been entered.
             if (self.seat.input_manager.inputAllowed(wlr_surface)) {
+                const wlr_seat = self.seat.wlr_seat;
+                const focus_change = wlr_seat.pointer_state.focused_surface != wlr_surface;
```

Comment:
could be moved into the `.normal` case below

Reply:
Good catch. This was leftover from before f597e7da63ed88b3e6156cd05867252b952d2094